ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/slot/SlotStoreAdaptor.java

```diff
@@ -0,0 +1,349 @@
+ * This program is distributed in the hope that it will be useful, but WITHOUT A
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * (C) 2020,
+ * @author Jonathan Halliday (jonathan.halliday@redhat.com)
+ */
+package com.arjuna.ats.internal.arjuna.objectstore.slot;
+
+import com.arjuna.ats.arjuna.common.Uid;
+import com.arjuna.ats.arjuna.exceptions.ObjectStoreException;
+import com.arjuna.ats.arjuna.logging.tsLogger;
+import com.arjuna.ats.arjuna.objectstore.ObjectStoreAPI;
+import com.arjuna.ats.arjuna.objectstore.StateStatus;
+import com.arjuna.ats.arjuna.state.InputObjectState;
+import com.arjuna.ats.arjuna.state.OutputObjectState;
+import com.arjuna.ats.internal.arjuna.common.UidHelper;
+import com.arjuna.common.internal.util.propertyservice.BeanPopulator;
+
+import java.io.IOException;
+import java.io.SyncFailedException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Adaptor class that wraps the SlotStore to make it look like an ObjectStore.
+ * Modelled on HornetqObjectStoreAdaptor.
+ *
+ * @author Jonathan Halliday (jonathan.halliday@redhat.com), 2020-03
+ */
+public class SlotStoreAdaptor implements ObjectStoreAPI {
+
+    private final SlotStore store;
+
+    // used for standalone bootstrap via StoreManager
+    public SlotStoreAdaptor() throws IOException {
+        SlotStoreEnvironmentBean envBean = BeanPopulator.getDefaultInstance(SlotStoreEnvironmentBean.class);
+        store = new SlotStore(envBean);
+    }
+
+    // used for beans wiring type bootstrap when running embedded.
+    public SlotStoreAdaptor(SlotStore slotStore) {
+        this.store = slotStore;
+    }
+
+    /**
+     * Obtain all of the Uids for a specified type.
+     *
+     * @param typeName       The type to scan for.
+     * @param foundInstances The object state in which to store the Uids
+     * @param matchState     The file type to look for (e.g., committed, shadowed). [StateStatus]
+     * @return <code>true</code> if no errors occurred, <code>false</code>
+     * otherwise.
+     */
+    @Override
+    public boolean allObjUids(String typeName, InputObjectState foundInstances, int matchState) throws ObjectStoreException {
+        if (tsLogger.logger.isTraceEnabled()) {
+            tsLogger.logger.trace("SlotStore.allObjUids(" + typeName + ", " + matchState + ")");
+        }
+
+        SlotStoreKey key = new SlotStoreKey(null, typeName, matchState);
+
+        SlotStoreKey[] matchingKeys = store.getMatchingKeys(key);
+
+        OutputObjectState buffer = new OutputObjectState();
+
+        try {
+            if (matchingKeys != null) {
```

Comment:
Superfluous test for null because matchingKeys cannot be null because SlotStoreAdaptor#getMatchingKeys(key) always returns a valid array.

Reply:
D'oh